script/hipify-inplace.sh

```diff
@@ -51,27 +60,47 @@ echo "Logs written to ${HIPIFY_LOG}"
 
 # Additional fixes for project-specific things and things hipify misses or gets
 # wrong.
-for src in ${srcs[@]}; do
-    sed -i 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' $src
-    sed -i 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' $src
-    sed -i 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' $src
-    sed -i 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' $src
-    sed -i 's@\bDGL_USE_CUDA\b@DGL_USE_ROCM@g' $src
-    sed -i 's@\bCUB_VERSION\b@HIPCUB_VERSION@g' $src
-    sed -i 's@\bCUDART_ZERO_BF16\b@HIPRT_ZERO_BF16@g' $src
-    sed -i 's@\bCUDART_INF_BF16\b@HIPRT_INF_BF16@g' $src
-    sed -i 's@\bthrust::cuda::par@thrust::hip::par@g' $src
-    sed -i 's@\b__nv_fp8_e4m3\b@__hip_fp8_e4m3@g' $src
-    sed -i 's@\b__nv_fp8_e5m2\b@__hip_fp8_e5m2@g' $src
-    sed -i 's@\bCUBLAS_GEMM_DEFAULT_TENSOR_OP\b@HIPBLAS_GEMM_DEFAULT@g' $src
-    sed -i 's@\bcurand4\b@hiprand4@g' $src
-    # hipify uses the old one
-    sed -i 's@\bhip_bfloat16\b@__hip_bfloat16@g' $src
-    sed -i 's@\b__trap();@abort();@' $src
+for src in ${all_srcs[@]}; do
+    sed -i -e 's@#include <hipblas.h>@#include <hipblas/hipblas.h>@' \
+        -e 's@#include <hipsparse.h>@#include <hipsparse/hipsparse.h>@' \
+        -e 's@#include <cuda_fp8.h>@#include <hip/hip_fp8.h>@' \
+        -e 's@#include <cuda_bf16.h>@#include <hip/hip_bf16.h>@' \
+	-e 's@#include <cub/cub.cuh>@#include <hipcub/hipcub.hpp>@' `# TODO: File issue` \
+	-e 's@#include <cuda/functional>@#include <cuda/std/functional>@' `# TODO: File issue` \
+	-e 's@#include "hip/extension/@#include "./cuda/extension/@' `# TODO: File issue` \
```

Comment:
Oh are these paths within graphbolt that are getting rewritten? I thought hipify was smarter than just rewriting every instance of "cuda" to "hip". I wonder why I didn't have any issues with that for my hipification of the main project

Reply:
Based on what I've seen, I think hipifying first normalizes includes and then does pattern matching. In `dgl/src`, the includes look like `#include "../../runtime/cuda/file"`, while graphbolt has [1 file](https://github.com/nod-ai/dgl/blob/51907e048558d49b21044ea5955d3d46466b1a72/graphbolt/src/python_binding.cc#L12) with lines like `#include "./cuda/file"`. After normalizing, it doesn't recognize it as a relative path.

Given it is only this one file, I could also just modify the source file and drop 3 of these sed lines.